### ISSUE TITLE
A false-positive: You should use assertTrue() instead of assertSame() when expecting "true"

### ISSUE THREAD

**Issue by kalessil** [phpstan/phpstan-phpunit#24]
Code sample: `$this->assertSame($content->published = true, $content->published());`

Since we have an assignment as the expected value arguments, assertTrue() can not be applied.

**Comment by ondrejmirtes** [comment]
Hi,
I'd argue that your test is doing too much - why couple your assertion with an assignment like this? 😊 Looks really weird. Since PHPStan rules for this see only the types passed and I don't want to complicate them because of this (It's really rare), I don't want to fix this. Thanks for understanding 😊

**Comment by kalessil** [comment]
This is testing that a model with lots of magic (`__set`, `__set`) has a working public API.
Might be weird, but that's how do you need to deal with those type of models.

No worries, I suppressed the message already. 

BTW, probably inline assignments support need to be improved. Can you share insights why it so complicated, so I know if you going to address the related issues in near future?

**Comment by ondrejmirtes** [comment]
I'd argue that your code would better like this 😊

```php
$content->published = true;
$this->assertTrue($content->published());
```

Currently, the rule for this (https://github.com/phpstan/phpstan-phpunit/blob/master/src/Rules/PHPUnit/AssertSameBooleanExpectedRule.php) works only with types (`$content->published = true` results in true - ConstantBooleanType). I could add an exception for assignment but it looks too edge-casey to me and I don't know how to generalize this for other situations, seems too rare 😊

**Comment by kalessil** [comment]
Probably =) I have some habits and a special love to inline assignments where my cognitive load reduces =)

Understood, type wise the case is correctly reported. Agreed, while complains are rare it makes no sense to address the case.
